test/unit/service/JobRoleServiceTests.ts

```diff
@@ -78,4 +88,55 @@ describe('JobRoleService', function () {
             }
         });
     });
+    describe('createJobRole', function () {
+        it('should return the created job role ID from response', async () => {
+            var mock = new MockAdapter(axios);
+    
+            const createdId = 1;
+            const createdJobRole = { ...jobroleAdd, id: createdId };
+    
+            mock.onPost(JobRoleService.URL, jobroleAdd).reply(200, createdJobRole);
+    
+            var resultId;
+            try {
+                resultId = await JobRoleService.createJobRole(jobroleAdd);
+                assert.strictEqual(resultId.id, createdId);
+            } catch (error) {
+                assert.fail('Unexpected error: ' + error.message);
+            }
+        });
+        it('should throw an error if the request fails', async () => {
+            var mock = new MockAdapter(axios);
+
+            mock.onPost(JobRoleService.URL, jobroleAdd).reply(500);
+
+            try {
+                await JobRoleService.createJobRole(jobroleAdd);
+               // assert.fail('Expected an error to be thrown.');
+            } catch (error) {
+                assert.strictEqual(error.message, 'Could not create job role.');
+            }
+        });
+        it('should throw an error if the job role is missing a required field', async () => {
```

Comment:
you're not mocking out your validator class so this test is also testing your validator class. please mock the validator and update tests for successful and unsuccessful validation